### PR TITLE
Build Python wheels for macOS

### LINKED ISSUE #133: pip install ctranslate2: no package found on macOS
Hi, 

Running `pip install ctranslate2` with the latest pip as per the installation instructions results in the following:
```
ERROR: Could not find a version that satisfies the requirement ctranslate2== (from versions: none)
ERROR: No matching distribution found for ctranslate2==
```

```
> pip --version
pip 20.0.2 from [...]/lib/python3.8/site-packages/pip (python 3.8)
```

```
> conda --version
conda 4.7.12
```

This is on macOS Mojave 10.14.6 (18G2022)

### PR DESCRIPTION
**Pull request by keichi**
Closes #133 

This PR allows building Python wheels for macOS. Embedded libctranslate2 is linked with Intel oneAPI MKL and DNN. Testing infrastructure for macOS is also setup.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -158,21 +182,34 @@ jobs:
...
+        if: startsWith(matrix.os, 'macos')
+        run: |
+          python -m pip install cibuildwheel==1.7.4
+          python -m cibuildwheel python --output-dir python/wheelhouse
+        env:
+          CIBW_BEFORE_ALL: python/tools/prepare_build_environment_macos.sh
+          CIBW_BEFORE_BUILD: pip install -r python/install_requirements.txt
+          CIBW_BEFORE_TEST: python/tools/prepare_test_environment.sh
+          CIBW_TEST_COMMAND: '{project}/python/tools/run_test.sh {project}'
+          CIBW_SKIP: cp27-* pp* *i686
```
Comment (guillaumekln) [inline .github/workflows/ci.yml:207]: Is there any way to factorize this block with the Linux step? Looks like only `CIBW_BEFORE_ALL` is different.

**Review by guillaumekln — COMMENTED** [review]
Thanks!

**Comment by keichi** [comment]
Thanks for your review, I think the CI script is more concise now.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (keichi) [reply .github/workflows/ci.yml]: I've used platform-specific variants instead.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -25,24 +26,44 @@ jobs:
...
           wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
           sudo apt-key add *.PUB
           sudo sh -c 'echo "deb https://apt.repos.intel.com/oneapi all main" > /etc/apt/sources.list.d/oneAPI.list'
           sudo apt-get update
 
+      - name: Install Intel oneAPI
+        if: startsWith(matrix.os, 'macos')
+        run: |
+          wget -q https://registrationcenter-download.intel.com/akdlm/irc_nas/17426/m_BaseKit_p_2021.1.0.2427_offline.dmg
+          hdiutil attach -noverify -noautofsck m_BaseKit_p_2021.1.0.2427_offline.dmg
```
[the remark this answers is not in the document]
Comment (keichi) [reply .github/workflows/ci.yml]: I've saved the URL to an environment variable.

**Comment by guillaumekln** [comment]
Thanks!

After reading the compilation logs, it seems OpenMP is not found on macOS:

```text
-- Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES) 
-- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES) 
-- Could NOT find OpenMP (missing: OpenMP_C_FOUND OpenMP_CXX_FOUND)
```

I'm not familiar with compilation on macOS. Should we install a separate package to enable compiling with OpenMP?

**Comment by keichi** [comment]
Hmm weird. On my local environment, CMake does find OpenMP. I'll look into it.

```
-- Found OpenMP_C: -Xclang -fopenmp (found version "3.1")
-- Found OpenMP_CXX: -Xclang -fopenmp (found version "3.1")
-- Found OpenMP: TRUE (found version "3.1")
```

**Comment by keichi** [comment]
@guillaumekln Ok fixed it. CMake was failing to detect OpenMP because the runtime included in MKL did not have the header file (`omp.h`). I installed LLVM's libomp to fix this but we still link to Intel's OpenMP runtime (`libiomp5`).

**Review by guillaumekln — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**